map.go

```diff
@@ -289,40 +289,40 @@ func (m *Map) Pin(fileName string) error {
 
 // LoadPinnedMap load a Map from a BPF file.
 //
-// Requires at least Linux 4.13, use LoadPinnedMapExplicit on
-// earlier versions.
+// Requires at least Linux 4.13, and is not compatible with
+// nested maps. Use LoadPinnedMapExplicit in these situations.
 func LoadPinnedMap(fileName string) (*Map, error) {
 	fd, err := getObject(fileName)
 	if err != nil {
 		return nil, err
 	}
-	spec, err := getMapSpecByFD(fd)
+	abi, err := newMapABIFromFd(fd)
 	if err != nil {
 		return nil, err
 	}
-	return newMap(fd, spec)
+	return newMap(fd, abi)
 }
 
 // LoadPinnedMapExplicit loads a map with explicit parameters.
-func LoadPinnedMapExplicit(fileName string, spec *MapSpec) (*Map, error) {
+func LoadPinnedMapExplicit(fileName string, abi *MapABI) (*Map, error) {
```

Comment:
What's the use case for this?

Reply:
Getting an ABI requires kernel 4.13 or something, so this lets users bypass that requirement.

It'd be nicer if we made the ABI "optional" but I think that would complicate the Map code a bit.